test/unit/Serializer/ArrayDtoSerializerTest.php

```diff
@@ -0,0 +1,548 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Unit\Serializer;
+
+use OnMoon\OpenApiServerBundle\Interfaces\Dto;
+use OnMoon\OpenApiServerBundle\Interfaces\ResponseDto;
+use OnMoon\OpenApiServerBundle\Serializer\ArrayDtoSerializer;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\ObjectType;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Operation;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Property;
+use OnMoon\OpenApiServerBundle\Types\ScalarTypesResolver;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Throwable;
+
+use function get_class;
+use function Safe\json_encode;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\Serializer\ArrayDtoSerializer
+ */
+class ArrayDtoSerializerTest extends TestCase
```

Comment:
Lines 117 and 125 are not covered
```php

if ($objectType !== null) {
    /** @psalm-suppress MissingClosureParamType */
    $converter = fn ($v) => $this->convert($deserialize, $v, $objectType);
}
...
if ($property->isArray()) {
    /** @psalm-suppress MissingClosureParamType */
    $converter = static fn ($v) => array_map(static fn ($i) => $converter($i), $v);
}
```

You can test them too by adding more cases to the dataproviders of the test methods.

First case happens when one of the properties is an object itself, and `convert` gets called recursively, it is enough to test one level of nesting when one recursive call happens.
Second case happens when one of the properties is an array of objects or scalaras.

These two cases should be checked for both request deserialization and response serialization.